src/edu/stanford/nlp/stats/Counters.java

```diff
@@ -1674,7 +1675,7 @@ private Counters() {} // only static methods
     C result = (C) c.getFactory().create();
     for (E key : c.keySet()) {
       double count = c.getCount(key);
-      double noise = -Math.log(1.0 - random.nextDouble()); // inverse of CDF for
+      double noise = -FastMath.log(1.0 - random.nextDouble()); // inverse of CDF for
```

Comment:
not sure if this is in any way critical enough to need a log1p as well

Reply:
Here the better solution is actually `-Math.log(random.nextDouble())`. Since the random values are supposedly uniform, it does not make sense to use 1- at all.

Reply:
random.nextDouble() can produce 0, right?  Perhaps that was the consideration.